File: 이코테_연습문제/greedy_number_card_game.py
# 숫자 카드 게임
n, m = map(int, input().split())
cards = []
for _ in range(n):
    cards.append(list(map(int, input().split())))

ans = 0
for row in cards:
    ans = max(min(row), ans)

print(ans)

File: 이코테_연습문제/GreedyNumberCardGame.java
import java.util.*;

public class GreedyNumberCardGame {
    public static void main(String[] args) {
        int n = 2;
        int m = 4;
        int[][] cards = {{7, 3, 1, 8},{3, 3, 3, 4}};
        int result = 0;
        
        for (int i=0; i < n; i++) {
            int row_min = Arrays.stream(cards[i]).min().orElse(0);
            result = Math.max(result, row_min);
        }

        System.out.println(result);
    }
}
